ti/report/parser/junit/gojunit/ingest.go

```diff
@@ -16,33 +16,48 @@ import (
 	ti "github.com/harness/ti-client/types"
 )
 
+const defaultRootSuiteName = "Root Suite"
+
+func getRootSuiteName(envs map[string]string) string {
+	if val, ok := envs["HARNESS_JUNIT_ROOT_SUITE_NAME"]; ok {
+		return val
+	}
+	return defaultRootSuiteName
+}
+
 // findSuites performs a depth-first search through the XML document, and
 // attempts to ingest any "testsuite" tags that are encountered.
-func findSuites(nodes []xmlNode, suites chan Suite) {
+func findSuites(nodes []xmlNode, suites chan Suite, parentFilename string, envs map[string]string) {
```

Comment:
You don't need to pass the full map. Just pass what you need (`parentFilename`)

Reply:
I need to pass rootSuiteName as well right?

Reply:
Or should I use a global variable for it?

Reply:
Discussed on call